cybersyn/scripts/central-planning.lua

```diff
@@ -69,6 +69,23 @@ function is_train_routable(e1, e2)
 	return (e1.surface == e2.surface)
 end
 
+---Checks that the train has a base schedule it can return to after a delivery.
+---If it does not, the train is removed from Cybersyn and an alert is generated.
+---@param train Train
+---@return boolean base_schedule_valid
+function validate_train_base_schedule(map_data, train_id, train)
+	local schedule = train.entity.get_schedule()
+	if schedule.get_record_count() > 0 then return true end
+
+	remove_train(map_data, train_id, train)
```

Comment:
I'm unsure if `remove_available_train` would be better for alerts. But then the train needs a new train.status, probably STATUS_TO_D.